bluemira/builders/EUDEMO/first_wall/wall.py

```diff
@@ -0,0 +1,231 @@
+"""
+import copy
+from typing import Any, Dict, List, Optional
+
+import bluemira.utilities.plot_tools as bm_plot_tools
+from bluemira.base.builder import BuildConfig, Component
+from bluemira.base.components import PhysicalComponent
+from bluemira.base.config import Configuration
+from bluemira.builders.shapes import OptimisedShapeBuilder
+from bluemira.display.palettes import BLUE_PALETTE
+from bluemira.geometry.parameterisations import PolySpline, PrincetonD
+from bluemira.geometry.wire import BluemiraWire
+
+
+class WallPolySpline(PolySpline):
+    """
+    Defines the geometry for reactor first wall, without a divertor,
+    based on the PolySpline parameterisation.
+    """
+
+    _defaults = {
+        "x1": {"value": 5.8},
+        "x2": {"value": 12.1},
+        "z2": {"value": 0},
+        "height": {"value": 9.3},
+        "top": {"value": 0.4},
+        "upper": {"value": 0.3},
+        "dz": {"value": -0.5},
+        "tilt": {"value": 0},
+        "lower": {"value": 0.5},
+        "bottom": {"value": 0.2},
+    }
+
+    def __init__(self, var_dict=None):
+        if var_dict is None:
+            var_dict = {}
+        defaults = copy.deepcopy(self._defaults)
+        defaults.update(var_dict)
+        super().__init__(defaults)
+
+        ib_radius = self.variables["x1"].value
+        ob_radius = self.variables["x2"].value
+        z2 = self.variables["z2"].value
+        height = self.variables["height"].value
+        top = self.variables["top"].value
+        upper = self.variables["upper"].value
+        dz = self.variables["dz"].value
+        tilt = self.variables["tilt"].value
+        lower = self.variables["lower"].value
+        bottom = self.variables["bottom"].value
+
+        self.adjust_variable(
+            "x1",
+            ib_radius,
+            lower_bound=ib_radius - 2,
+            upper_bound=ib_radius * 1.1,
+        )
+        self.adjust_variable(
+            "x2",
+            value=ob_radius,
+            lower_bound=ob_radius * 0.9,
+            upper_bound=ob_radius + 2,
+        )
+        self.adjust_variable("z2", z2, lower_bound=-0.9, upper_bound=0.9)
+        self.adjust_variable(
+            "height", height, lower_bound=height - 0.001, upper_bound=50
+        )
+        self.adjust_variable("top", top, lower_bound=0.05, upper_bound=0.75)
+        self.adjust_variable("upper", upper, lower_bound=0.2, upper_bound=0.7)
+        self.adjust_variable("dz", dz, lower_bound=-5, upper_bound=5)
+        self.adjust_variable("tilt", tilt, lower_bound=-25, upper_bound=25)
+        self.adjust_variable("lower", lower, lower_bound=0.2, upper_bound=0.7)
+        self.adjust_variable("bottom", bottom, lower_bound=0.05, upper_bound=0.75)
+
+        # Fix 'flat' to avoid drawing the PolySpline's outer straight.
+        # The straight is often optimised to near-zero length, which
+        # causes an error when CAD tries to draw it
+        self.fix_variable("flat", 0)
+
+
+class WallPrincetonD(PrincetonD):
```

Comment:
I've ummed and ahhed a bit over these builder-specific shapes. On one hand I like that they are tuned to the details of the build, but in principle this could be done from the configuration and tuned there... perhaps we can leave this in and see if it works well as an approach (particularly if the bounds need to be relative (if they do, then it could be worth us considering allowing derived replacements in the config)).

Reply:
spun out in #765